tests/union_vector/union_vector_generated.ts

```diff
@@ -426,4 +542,109 @@ static createMovie(builder:flatbuffers.Builder, mainCharacterType:Character, mai
   Movie.addCharacters(builder, charactersOffset);
   return Movie.endMovie(builder);
 }
+
+/**
+ * @returns MovieT
+ */
+unpack(): MovieT {
+  return new MovieT(
+    this.mainCharacterType(),
+    (() => {
+    let targetEnumStr = Character[this.mainCharacterType()];
+    if(targetEnumStr === 'NONE') { return null; } 
+
+    if(targetEnumStr === 'MuLan') { return (this.mainCharacter(new Attacker())! as Attacker).unpack();}
```

Comment:
Does TS have a switch that works on strings?
Would it be good to stick this function in an actual named function? if a schema has multiple fields of the same union type, then all this code will be duplicated.